Готова: Проектная работа 5 «Карты подскажут», спринт 3

vladimirton opened the pull request:
- Добавлены переменные тем и управление цветовой схемой
- Настроены цвета текста и фона для каждой темы
- Реализованы интерактивные кнопки переключения тем
- Добавлены специальные стили для элементов темной темы
- Настроена адаптивная типографика и макет

vladimirton committed:
feat: реализация переключателя темной/светлой темы с адаптивными стилями

- Добавлены переменные тем и управление цветовой схемой
- Настроены цвета текста и фона для каждой темы
- Реализованы интерактивные кнопки переключения тем
- Добавлены специальные стили для элементов темной темы
- Настроена адаптивная типографика и макет

Avaakum commented on index.html:
```diff
@@ -12,11 +12,154 @@
+          <a
+            href="https://play.google.com/store/apps/details?id=com.monoloco.obliquestrategies&hl=en_US&gl=US&pli=1"
+            target="_blank"
+            >для Android</a
```
для Android  ->   Android

Avaakum commented on index.html:
```diff
@@ -12,11 +12,154 @@
+        <p class="content__paragraph">
+          И кстати, сейчас необязательно охотиться за редкими физическими
+          колодами Oblique Strategies — тем более, что их моментально разбирают
+          фанаты Ино. Есть довольно симпатичные веб-сайты с
```
Слово "веб-сайты" так же должно быть ссылкой из макета

Avaakum commented on styles/style.css:
```diff
@@ -20,3 +37,225 @@ li,
+}
+
+.heading {
+  font-family: var(--accent-font, serif);
```
---

**Отлично:**

Корректная альтернатива

Avaakum commented on index.html:
```diff
@@ -12,11 +12,154 @@
+    <header class="header">
+      <h1 class="heading header_title content-section">Карты подскажут</h1>
+      <nav class="header__menu content-section">
+        <div class="header__logo">
```
---

**Можно лучше:**

Логотип должен быть ссылкой **без атрибута href,**(что бы по ссылке не было перехода). А так же, что бы при навигации с клавиатуры этот элемент не получал видимый фокус. **Пустой атрибут href тоже нужно удалить.**

Avaakum commented on index.html:
```diff
@@ -12,11 +12,154 @@
+        </p>
+      </div>
+      <figure class="content__figure">
+        <picture>
```
---

**Отлично:**

Отличная семантика и структура

Avaakum commented on index.html:
```diff
@@ -12,11 +12,154 @@
+        <p class="content__paragraph">
+          Вот что сам Брайан Ино рассказывает о создании Oblique Strategies:
+        </p>
+        <blockquote class="content__blockquote">
```
---

**Отлично:**

Отличная семантика

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Доброго времени суток. У вас получилась хорошая работа!

_Что понравилось:_

- работают темы
- отличная структура файлов проекта
- Хорошая семантика
- Хорошие стили 
- Отзывчивая верстка
- Похоже на макет на брейк-поинтах


_Что необходимо исправить:_


- **Все комментарии, кроме комментариев с пометками «Можно лучше» и «Отлично». А так же не забудьте проверить, есть ли свёрнутые комментарии, которые обычно находятся по центру страницы с ревью при прокрутке вниз, и выглядят следующим образом https://disk.yandex.ru/i/FyX-8tzeqLjimw**

- Для ссылки нужно присвоить ховер из макета. Подчёркивание должно быть с прозрачностью при ховере.



_Что можно улучшить:_

- Все комментарии с пометкой "Можно лучше"
- Для всех функций ` var() ` в CSS нужно задать резервные значения,  на случай если доступ к кастомному свойству по имени будет невозможен. 
- Можно использовать атрибуты `target="_blank"` для ссылок на сторонние ресурсы для того чтобы пользователю не приходилось заново открывать сайт проекта
- больше использовать currentColor, где это возможно
- больше использовать vi вместо vw


Желаю успеха!

vladimirton committed:
fix: улучшение стилей и доступности

- Добавлены резервные значения для CSS-переменных через var()
- Добавлен hover-эффект для ссылок с полупрозрачным подчёркиванием
- Заменены vw на vi для более корректного масштабирования
- Добавлены атрибуты target="_blank" для внешних ссылок
- Оптимизировано использование currentColor
- Улучшена доступность логотипа через обёртку в <a>

Avaakum reviewed (approved):
Развернуть общий комментарий ⬇︎

Здравствуйте! Рад вам сообщить, что ваша работа принята. Сейчас страничка выглядит прекрасно, всё работает отлично. Отличная семантика и структура. Желаю вам удачи в следующих спринтах и проектах! 




Ссылки:

- Можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`. Но если страница с адаптацией, то всегда стоит проверять как текст с пробелами выглядит на всех ширинах.
- Про альтернативный текст для картинок https://habr.com/ru/post/341810/
- [Ссылка про подгрузку шрифтов](https://only-to-top.ru/blog/tools/2019-03-13-nastrojte-pokaz-vsego-teksta-vo-vremya-zagruzki-veb-shriftov.html) 


Несколько  ссылок на материал по теме нативных CSS переменных от авторов практикума:

- [1 часть](https://tproger.ru/articles/custom-properties-in-css-part-1/)

- [2 часть](https://tproger.ru/articles/custom-properties-in-css-part-2/)

- [3 часть](https://tproger.ru/articles/custom-properties-in-css-part-3/)



**Hover и псевдоэлементы**

Псевдоэлементы:

- https://developer.mozilla.org/ru/docs/Web/CSS/Pseudo-elements
- https://habr.com/ru/post/154319/
-  https://webformyself.com/krutoe-ispolzovanie-psevdoelementov-css-before-i-after/
-  https://www.internet-technologies.ru/articles/5-priemov-css3-s-psevdoelementami-before-i-after.html

Hover:

- Базовая спека http://htmlbook.ru/css/hover		
- https://codepen.io/igosheva_nastya/pen/qeJmMM

**БЭМ**

- Соглашение по именованию БЭМ https://ru.bem.info/methodology/naming-convention/
- Часто используемые слова в классах https://github.com/yoksel/common-words
- Где описывать геометрию https://ru.bem.info/methodology/faq/#Зачем-внешнюю-геометрию-и-позиционирование-задавать-через-родительский-блок
- Проверить БЭМ валидатором https://nglazov.github.io/bem-validator-page/

** Доступность**

- https://medium.com/web-standards/what-accessibility-taught-me-207d1ff0e80c



**Игры по flex/grid**

- https://flexboxfroggy.com/#ru
- https://cssgridgarden.com/#ru




**Grid-layout**

- https://css-tricks.com/snippets/css/complete-guide-grid/ (en)
- https://tuhub.ru/posts/css-grid-complete-guide
- https://medium.com/@stasonmars/вёрстка-на-grid-в-css-полное-руководство-и-справочник-220508316f8b
